hatch/src/cli/commands/new.rs

```diff
@@ -39,16 +39,14 @@ impl<'new> New {
     if args.is_present(TYPE) {
       let type_arg: String = value_t!(args, TYPE, String).unwrap();
 
-      // we cannot use the static variables BIN, STATIC, or SHARED because it is illegal in Rust
-      // to pattern match on a static value
-      let kind = match type_arg.as_ref() {
-        "bin" => ProjectKind::Binary,
-        "static" => ProjectKind::Library(LibraryKind::Static),
-        "shared" => ProjectKind::Library(LibraryKind::Shared),
-        _ => ProjectKind::Library(LibraryKind::Static),
-      };
-
-      kind
+      // we cannot use the static variables BIN, STATIC, or SHARED directly because it is illegal in
+      // Rust to pattern match on a static variable
+      match type_arg.as_str() {
+        value if (value == BIN) => ProjectKind::Binary,
+        value if (value == STATIC) => ProjectKind::Library(LibraryKind::Static),
+        value if (value == SHARED) => ProjectKind::Library(LibraryKind::Shared),
```

Comment:
Are the parenthesis required in `value if (value == FOO) => ... `? If they are not, I think we should omit them. I like the more terse syntax here. I'm open to counters.

Reply:
I personally am a fan of the parenthesis.  I think that with the arrow syntax right after is just too much.  This leads to better visualization of the semantics of the statement.

Reply:
Oh lets get rid of them!